CMSgov/design-system
# [WNMGDS-969] Typescript Dialog Conversion

## Summary
Converted the Dialog component to Typescript

## How to test
`yarn test`, `yarn build && yarn start` each system to confirm docs site, `yarn storybook` to confirm storybook & `yarn type-check' to confirm no ts typing issues.
